### PR TITLE
Fix width and margins of the notebook footer.

### PR DESCRIPTION
**Pull request by HaudinFlorence**
This PR fixes the width and the margins of the notebook footer.

## References

This PR should fix issue #16380 
## Code changes

The proposes changes only concerns the style of the notebook footer and are in `packages/notebook/style/notebookfooter.css`

### CODE REVIEW

**Comment by HaudinFlorence** [comment]
bot please update snapshots

File: galata/test/documentation/general.test.ts-snapshots/command-palette-documentation-linux.png
Comment (krassowski) [inline galata/test/documentation/general.test.ts-snapshots/command-palette-documentation-linux.png:1]: These snapshot changes appear unrelated and are causing the tests to fail I think:

```
Notice:   3 failed
    [documentation] › test/documentation/general.test.ts:618:7 › General › Command Palette ─────────
    [documentation] › test/documentation/internationalization.test.ts:14:7 › Internationalization › Menu 
    [documentation] › test/documentation/internationalization.test.ts:28:7 › Internationalization › Confirm language 
```

**Review by krassowski — COMMENTED** [review]
Thank you for working on a fix! It looks good in the (now default) full windowing mode:

Screenshot from 2024-05-28 20-55-37

but in the `none` mode it seems that the spacing is too much:

Screenshot from 2024-05-28 20-55-27

File: packages/notebook/style/notebookfooter.css [hunk 1]
```diff
@@ -5,20 +5,20 @@
...
     100% -
       (
-        var(--jp-cell-prompt-width) + var(--jp-cell-collapser-width) +
-          var(--jp-cell-padding) + var(--jp-cell-padding)
+        var(--jp-cell-prompt-width) + var(--jp-cell-collapser-width) + 2 *
+          var(--jp-cell-padding) + 2 * var(--jp-notebook-padding)
       )
   );
-  border: var(--jp-border-width) solid var(--jp-cell-editor-border-color);
+  border: 0.8px solid var(--jp-cell-editor-border-color);
```
[the remark this answers is not in the document]
Comment (HaudinFlorence) [reply packages/notebook/style/notebookfooter.css:20]: @krassowski I can revert this change.

File: galata/test/documentation/general.test.ts-snapshots/command-palette-documentation-linux.png
Comment (HaudinFlorence) [reply galata/test/documentation/general.test.ts-snapshots/command-palette-documentation-linux.png:1]: Do you mean that I should take back the snapshots from the main branch ?

Comment (krassowski) [reply galata/test/documentation/general.test.ts-snapshots/command-palette-documentation-linux.png:1]: Yes, I think that after that these tests should pass. If not, then we need to investigate (it looks like maybe different runners get different variants of Chinese fonts?)

**Comment by krassowski** [comment]
@HaudinFlorence can you test the changes I pushed?

**Comment by HaudinFlorence** [comment]
> @HaudinFlorence can you test the changes I pushed?

@krassowski Thanks for working on this PR.  In the none mode, the width and left margin are ok but the footer remains at it initial position when adding new cells:
file:///home/fhaudin/Videos/Screencasts/Screencast%20from%2005-06-2024%2017:26:31.webm

Note also that in both modes, the top margin of the footer is slightly too large

**Comment by krassowski** [comment]
Thank you for this recording! I saw this happen earlier on CI (https://github.com/jupyterlab/jupyterlab/issues/14385); I now realise what is happening - the button is positioned below the spacer `jp-WindowedPanel-inner` element, which has either:
- managed height when `full` windowing mode is active
- unset height when `defer` or `none` is used

I did not see the issue you pointer out in testing because I was reloading JupyterLab or opening a new notebook after changing settings. I think that you did not reload JupyterLab after the change and this revealed a bug in that the previously set `height` of `jp-WindowedPanel-inner` remained set after switching to `none` but it was no longer managed so it stuck on an incorrect value.

This should be fixed by aafff6e1409aff45e585a1a0f10dcf14b61c8bec.

> Note also that in both modes, the top margin of the footer is slightly too large

They seem fine to me. If there was any margin added explicitly I would accommodate your suggestion by reducing it, but since there is no explicit `margin-top` anymore I think maybe we can leave it as is?

**Review by krassowski — APPROVED** [review]

**Comment by krassowski** [comment]
@meeseeksdev please backport to 4.2.x

**Comment by krassowski** [comment]
FYI I opened two follow-up issues for edge cases:
- https://github.com/jupyterlab/jupyterlab/issues/16491
- https://github.com/jupyterlab/jupyterlab/issues/16492
